src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -843,15 +843,31 @@ void Terminal::_AdjustCursorPosition(const COORD proposedPosition)
         }
     }
 
-    if (updatedViewport)
+    // If the viewport moved, or we circled the buffer, we might need to update
+    // our _scrollOffset
+    if (updatedViewport || newRows != 0)
     {
+        const auto oldScrollOffset = _scrollOffset;
+
         // scroll if...
         //   - no selection is active
         //   - viewport is already at the bottom
         const bool scrollToOutput = !IsSelectionActive() && _scrollOffset == 0;
 
         _scrollOffset = scrollToOutput ? 0 : _scrollOffset + scrollAmount + newRows;
 
+        // Clamp the range to make sure that we don't scroll way off the top of the buffer
+        _scrollOffset = std::clamp(_scrollOffset,
+                                   0,
+                                   _buffer->GetSize().Height() - _mutableViewport.Height());
+
+        // If the new scroll offset is different, then we'll still want to raise a scroll event
+        updatedViewport |= oldScrollOffset != _scrollOffset;

```

Comment:
augh binary or on a logical expression
technically works
is technically incorrect

Reply:
```suggestion
        updatedViewport = updatedViewport || (oldScrollOffset != _scrollOffset);
```